python/ray/_private/package.py

```diff
@@ -0,0 +1,276 @@
+    """This function is used to create a package file based on working directory
+    and python local modules.
+
+    Args:
+        pkg_file (pathlib.Path): The path of file to be created.
+        working_dir (str): The working directory.
+        modules (list[module]): The python modules to be included.
+
+    Returns:
+        None
+    """
+    with ZipFile(pkg_file, "w") as zip_handler:
+        if working_dir:
+            # put all files in /path/working_dir into zip
+            working_path = Path(working_dir)
+            _zip_module(working_path, working_path, zip_handler)
+        for module in modules or []:
+            logger.info(module.__file__)
+            # we only take care of modules with path like this for now:
+            #    /path/module_name/__init__.py
+            # module_path should be: /path/module_name
+            module_path = Path(module.__file__).parent
+            _zip_module(module_path, module_path.parent, zip_handler)
+
+
+def fetch_package(pkg_uri: str, pkg_file: Path):
+    """This function is used to fetch a pacakge from the given uri to local
+    filesystem.
+
+    Args:
+        pkg_uri (str): The uri of the package to download.
+        pkg_file (pathlib.Path): The path in local filesystem to download the
+            package.
+
+    Returns:
+        The number of bytes downloaded.
+    """
+    (protocol, pkg_name) = _parse_uri(pkg_uri)
+    if protocol in (Protocol.GCS, Protocol.PIN_GCS):
+        code = internal_kv._internal_kv_get(pkg_uri)
+        code = code or b""
+        pkg_file.write_bytes(code)
+        return len(code)
+    else:
+        raise NotImplementedError(f"Protocol {protocol} is not supported")
+
+
+def _store_package_in_gcs(gcs_key: str, data: bytes):
+    internal_kv._internal_kv_put(gcs_key, data)
+    return len(data)
+
+
+def push_package(pkg_uri: str, pkg_file: Path):
+    """This function is to push a local file to remote uri. Right now, only GCS
+    is supported.
+
+    Args:
+        pkg_uri (str): The uri of the package to upload to.
+        pkg_file (Path): Path of the local file.
+
+    Returns:
+        The number of bytes uploaded.
+    """
+    (protocol, pkg_name) = _parse_uri(pkg_uri)
+    data = pkg_file.read_bytes()
+    if protocol in (Protocol.GCS, Protocol.PIN_GCS):
+        _store_package_in_gcs(pkg_uri, data)
+    else:
+        raise NotImplementedError(f"Protocol {protocol} is not supported")
+
+
+def package_exists(pkg_uri: str):
+    """Check whether the package with given uri exists or not.
+
+    Args:
+        pkg_uri (str): The uri of the package
+
+    Return:
+        True for package existing and False for not.
+    """
+    (protocol, pkg_name) = _parse_uri(pkg_uri)
+    if protocol in (Protocol.GCS, Protocol.PIN_GCS):
+        return internal_kv._internal_kv_exists(pkg_uri)
+    else:
+        raise NotImplementedError(f"Protocol {protocol} is not supported")
+
+
+def rewrite_working_dir_uri(job_config: JobConfig):
```

Comment:
```suggestion
def rewrite_working_dir_uri(job_config: JobConfig) -> None
```

Also consider returning the rewritten JobConfig instead of mutating it in-place.

Reply:
Do you mean to return a new JobConfig instead?